tests/behat/src/FixtureContext.php

```diff
@@ -0,0 +1,281 @@
+use SilverStripe\BehatExtension\Utility\StepHelper;
+use SilverStripe\Versioned\ChangeSet;
+use SilverStripe\Versioned\ChangeSetItem;
+
+/**
+ * Context used to create fixtures in the SilverStripe ORM.
+ */
+class FixtureContext extends BaseFixtureContext
+{
+    use StepHelper;
+
+    /**
+     * @Then /^I should see the campaign "([^"]+)"$/
+     * @param string $name
+     */
+    public function iShouldSeeTheCampaign($name)
+    {
+        $item = $this->getCampaign($name);
+
+        assertNotNull($item, "Could not find campaign \"$name\" in the list");
+    }
+
+    /**
+     * @Then /^I should not see the campaign "([^"]+)"$/
+     * @param string $name
+     */
+    public function iShouldNotSeeTheCampaign($name)
+    {
+        $item = $this->getCampaign($name);
+
+        assertNull($item, "Found campaign \"$name\" in the list");
+    }
+
+    /**
+     * @Then /^I should see "([^"]+)" in column (\d+) of the "([^"]+)" campaign$/
+     * @param $name
+     * @param $column
+     */
+    public function iShouldSeeInTheCampaignColumn($text, $column, $name)
+    {
+        $campaignRow = $this->getCampaign($name);
+        assertNotNull($campaignRow, sprintf('Could not find campaign "%s"', $name));
+
+        $cell = $campaignRow->find('xpath', "/td[{$column}][contains(text(), '{$text}')]");
+
+        assertNotNull($cell, sprintf('Could not find "%s" in column %d of "%s"', $text, $column, $name));
+    }
+
+    /**
+     * @Given /^I view the campaign "([^"]*)"$/
+     * @param $name
+     */
+    public function iViewTheCampaign($name)
+    {
+        $item = $this->getCampaign($name);
+        assertNotNull($item, sprintf('Campaign %s not found', $name));
+
+        $item->click();
+    }
+
+    /**
+     * @Given /^I edit the campaign "([^"]*)"$/
+     * @param $name
+     */
+    public function iEditTheCampaign($name)
+    {
+        $item = $this->getCampaign($name);
+        assertNotNull($item, sprintf('Campaign %s not found', $name));
+
+        $button = $item->find('css', '.font-icon-cog');
+        assertNotNull($button, sprintf('Campaign %s has no edit button', $name));
+        $button->click();
+    }
+
+    /**
+     * @Given /^I delete the campaign "([^"]*)"$/
+     * @param $name
+     */
+    public function iDeleteTheCampaign($name)
+    {
+        $item = $this->getCampaign($name);
+        assertNotNull($item, sprintf('Campaign %s not found', $name));
+
+        $button = $item->find('css', '.font-icon-cancel');
+        assertNotNull($button, sprintf('Campaign %s has no delete button', $name));
+        $button->click();
+
+        /*@ var Behat\Mink\Driver\Selenium2Driver $driver */
+        $driver = $this->getMainContext()->getSession()->getDriver();
+        $driver->getWebDriverSession()->accept_alert();
+    }
+
+    /**
+     * @Then /^I should see the "([^"]*)" form$/
+     * @param string $id HTML ID of form
+     */
+    public function iShouldSeeTheForm($id)
```

Comment:
Can use `SilverStripe\AssetAdmin\Tests\Behat\Context\FixtureContext::iShouldSeeTheForm` instead of duplicating, after adding asset admin to require-dev.

Reply:
I tried this, but it actually doesn't work, because they both inherit from `SilverStripe\BehatExtension\Context\FixtureContext`, so there are all kinds of collisions on the inherited methods, e.g:

```
      SilverStripe\CampaignAdmin\Tests\Behat\Context\FixtureContext::assertMessageBoxContainsText()
      SilverStripe\AssetAdmin\Tests\Behat\Context\FixtureContext::assertMessageBoxContainsText()
    And I click "More options" in the "#ActionMenus" element
      Step "/^I should see "(?P<text>(?:[^"]|\\")*)" in the message box$/" is already defined in SilverStripe\CampaignAdmin\Tests\Behat\Context\FixtureContext::assertMessageBoxContainsText()
```